extract_duplicate_rows.py

```diff
@@ -0,0 +1,119 @@
+import argparse
+import pandas as pd
+from pathlib import Path
+
+
+def _get_args() -> argparse.Namespace:
+    """Returns the command-line arguments for this program."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--data_file",
+        help="Path to XLSX spreadsheet provided by FTVA",
+        required=True,
+    )
+    parser.add_argument(
+        "--tapes_tab_name",
+        help="Name of the spreadsheet tab containing tape data",
+        default="Tapes(row 4560-24712)",
+    )
+    parser.add_argument(
+        "--output_duplicate_path",
+        help="Path to the output spreadsheet for duplicate rows",
+        default="duplicate_rows.xlsx",
+    )
+    parser.add_argument(
+        "--remove_duplicates",
+        help="If specified, remove duplicate rows from the spreadsheet in place",
+        action="store_true",
+    )
+    args = parser.parse_args()
+    return args
+
+
+def _find_duplicate_rows(df: pd.DataFrame) -> pd.DataFrame:
+    """Find duplicate rows in the DataFrame based on the 'Legacy Path' column."""
+    # Check if 'Legacy Path' column exists
+    if "Legacy Path" not in df.columns:
+        raise ValueError("The DataFrame does not contain a 'Legacy Path' column.")
+
+    # Find duplicate rows based on the 'Legacy Path' column
+    duplicate_rows = df[df.duplicated(subset=["Legacy Path"], keep=False)]
+
+    return duplicate_rows
+
+
+def _format_duplicate_rows(duplicate_rows: pd.DataFrame) -> pd.DataFrame:
+    """Format duplicate rows for better readability in an output spreadsheet."""
+    # Create a copy of the DataFrame to avoid modifying the original slice,
+    # which causes pandas to raise a SettingWithCopyWarning
+    duplicate_rows = duplicate_rows.copy()
+    # Add a new column with the original row index
+    duplicate_rows.reset_index(inplace=True)
+    # increment each index by 2 to match the original spreadsheet
+    # (1 based index, plus header row)
+    duplicate_rows["index"] += 2
+    # rename the index column
+    duplicate_rows.rename(columns={"index": "Original Row Number"}, inplace=True)
+
+    return duplicate_rows
+
+
+def _remove_duplicates_from_df(df: pd.DataFrame, tapes_tab_name: str) -> pd.DataFrame:
+    """Remove duplicate rows from the DataFrame."""
+    # Remove duplicates based on 'Legacy Path' column, keeping the first occurrence
+    # and dropping the rest
+    df.drop_duplicates(subset=["Legacy Path"], keep="first", inplace=True)
+    return df
+
+
+def _remove_duplicates_from_spreadsheet(
+    df: pd.DataFrame, data_file_path: Path, tapes_tab_name: str
+) -> None:
+    """Remove duplicate rows from the DataFrame and save it back to the spreadsheet."""
+    # Remove duplicates from the DataFrame
+    df = _remove_duplicates_from_df(df, tapes_tab_name)
+    # Save the cleaned DataFrame back to the spreadsheet
+    # only overwrite the specified tab, and keep the rest of the spreadsheet intact
```

Comment:
Nitpick:  The 2 lines of this comment seem to belong together, since no punctuation at end of 1st line and the 2nd line starts with lower case, but it's really 2 statements.  Either end line 1 with semicolon, or start line 2 with `Only` instead of `only`. :pedant_face: